app/containers/wallet/dialogs/WalletTrezorConnectDialogContainer.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import React, { Component } from 'react';
+import { observer, inject } from 'mobx-react';
+
+import environment from '../../../environment';
+import type { InjectedDialogContainerProps } from '../../../types/injectedPropsType';
+
+import AboutDialog from '../../../components/wallet/trezorConnect/AboutDialog';
+import ConnectDialog from '../../../components/wallet/trezorConnect/ConnectDialog';
+import SaveDialog from '../../../components/wallet/trezorConnect/SaveDialog';
+
+import { ProgressStepOption } from '../../../stores/ada/TrezorConnectStore';
+
+type Props = InjectedDialogContainerProps;
+@inject('stores', 'actions') @observer
+export default class WalletTrezorConnectDialogContainer extends Component<Props> {
+
+  static defaultProps = { actions: null, stores: null, children: null, onClose: () => {} };
+
+  cancel = () => {
+    this.props.onClose();
+    this._getAction().cancel.trigger();
+  };
+
+  render() {
+    const store = this._getStore();
+    const action = this._getAction();
+
+    let component = null;
+
+    switch (store.progressInfo.currentStep) {
```

Comment:
I would move this logic outside the render in a new function

Reply:
good at it is